components/brave_origin/brave_origin_service.cc

```diff
@@ -70,61 +88,50 @@ bool BraveOriginService::SetBrowserPolicyValue(std::string_view pref_name,
   }
 
   // Set the policy value in BraveOriginPolicyManager
-  manager->SetBrowserPolicyValue(pref_name, value);
-
-  // Also set the corresponding pref value in local_state
-  // If not user-settable and value equals default, clear the pref instead
-  // We don't want a user to have a value set in local state if the value is not
-  // user settable as they would be stuck with that value if they stopped being
-  // a Brave Origin user
-  if (!pref_info->user_settable && value == pref_info->default_value) {
-    local_state_->ClearPref(pref_info->pref_name);
-  } else {
-    local_state_->SetBoolean(pref_info->pref_name, value);
+  PrefService* target_prefs = nullptr;
+  if (manager->IsBrowserPolicy(pref_info->pref_name)) {
+    manager->SetPolicyValue(pref_name, value);
```

Comment:
What do you think about having `SetPolicyValue` detect whether the policy is browser or profile-based internally, so callers don't have to call `IsBrowserPolicy`/`IsProfilePolicy`?